common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -61,22 +61,29 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
   }
 
   def cleanAmpEmbed(document: Document) = {
+    document.getElementsByClass("element-embed")
+      .filter(_.getElementsByTag("iframe").length != 0)
+      .foreach(_.getElementsByTag("iframe").map(_.remove))
+  }
 
-    document.getElementsByClass("element-embed").filter { element: Element =>
-      element.getElementsByTag("iframe").length != 0
-    }.foreach { embed: Element =>
-      embed.getElementsByTag("iframe").map { element: Element =>
-        val src = element.attr("srcdoc") // TODO it's a hack searching through the doc but CAPI doesn't have the shortcode yet
+  def cleanAmpInstagram(document: Document) = {
+    document.getElementsByClass("element-instagram").foreach { embed: Element =>
+      embed.getElementsByTag("a").map { element: Element =>
+        val src = element.attr("href")
         val list = src.split("""instagram\.com/p/""")
```

Comment:
To clarify my other comment:

    document.getElementsByClass("element-instagram")
      .foreach(_.getElementsByTag("a").map {
        ...
      })